### ISSUE TITLE
improperly sorted hostlist in 'flux resource list' output

### ISSUE THREAD

**Issue by ryanday36** [flux-framework/flux-core#5977]
the hostlist for allocated nodes in the `flux resource list` output doesn't appear to have been properly sorted and compacted. E.g.
 on corona:
```
[day36@corona212:~]$ flux resource list | grep allocated
 allocated pbatch         75     3600      600 corona[187-188,192-203,205,204,206-207,213-220,222,221,223-243,245,244,246-256,259,263-267,278-280,282-283,285-286]
[day36@corona212:~]$ hostlist --sort corona[187-188,192-203,205,204,206-207,213-220,222,221,223-243,245,244,246-256,259,263-267,278-280,282-283,285-286]
corona[187-188,192-207,213-256,259,263-267,278-280,282-283,285-286]
[day36@corona212:~]$
```
or rzvernal:
```
[day36@rzvernal10:~]$ flux resource list | grep allocated
 allocated pdebug         14      896      112 rzvernal[12-14,16,15,25,27,20,28-29,31,34,36,39]
[day36@rzvernal10:~]$
```
other states (down, free) seem okay. This just seems to be the allocated nodes. I spot checked node to rank mappings with `flux overlay lookup` on corona, and those looked okay to me.

**Comment by grondo** [comment]
This is a known issue (fixed in 0.62.0): #5885.

Unfortunately, it also means the hostnames can actually be incorrect... :grimacing:

**Comment by grondo** [comment]
Closing as a duplicate of #5885. Please reopen if we're still seeing any issues with 0.62.0.
